src/map/utils/petutils.cpp

```diff
@@ -1306,7 +1306,11 @@ namespace petutils
                     }
                 }
             }
-
+	     CCharEntity* PChar = (CCharEntity*)PMaster;

```

Comment:
you need to check if the PMaster is a character: C-style casts don't check this for you, it'll just crash or do other horrible things

if (PMaster->objtype == TYPE_PC)